Guava 24.0 Multiset incompatibility

Using Equalsverifier 2.4.2 and Guava 24.0, tests of classes which (indirectly) reference a `Multiset` fail. The issue doesn't happen with Guava 23.6, so the changes in https://github.com/google/guava/compare/v23.6...v24.0 have something to do with it.

The issue can also be reproduced using the Equalsverifier build itself. Just change:
```diff
diff --git a/pom.xml b/pom.xml
index eb96f3ca..fa447f05 100644
--- a/pom.xml
+++ b/pom.xml
@@ -140,7 +140,7 @@ limitations under the License.
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>20.0</version> <!-- Don't upgrade: version 21.0 and up require Java 8 -->
+            <version>24.0-jre</version> <!-- Don't upgrade: version 21.0 and up require Java 8 -->
             <scope>test</scope>
         </dependency>
         <dependency>
```
Then when running `mvn clean install` we see:
```
[ERROR] succeed_whenClassUsesGoogleGuavaMultiset(nl.jqno.equalsverifier.integration.extended_contract.ExternalApiClassesTest)  Time elapsed: 0.031 s  <<< FAILURE!
java.lang.AssertionError: 
ReflectionException: java.lang.IllegalAccessException: Class nl.jqno.equalsverifier.internal.prefabvalues.factories.AbstractReflectiveGenericFactory can not access a member of class com.google.common.collect.AbstractMultiset with modifiers "public final"
For more information, go to: http://www.jqno.nl/equalsverifier/errormessages
	at nl.jqno.equalsverifier.integration.extended_contract.ExternalApiClassesTest.succeed_whenClassUsesGoogleGuavaMultiset(ExternalApiClassesTest.java:40)
Caused by: nl.jqno.equalsverifier.internal.exceptions.ReflectionException: java.lang.IllegalAccessException: Class nl.jqno.equalsverifier.internal.prefabvalues.factories.AbstractReflectiveGenericFactory can not access a member of class com.google.common.collect.AbstractMultiset with modifiers "public final"
	at nl.jqno.equalsverifier.integration.extended_contract.ExternalApiClassesTest.succeed_whenClassUsesGoogleGuavaMultiset(ExternalApiClassesTest.java:40)
Caused by: java.lang.IllegalAccessException: Class nl.jqno.equalsverifier.internal.prefabvalues.factories.AbstractReflectiveGenericFactory can not access a member of class com.google.common.collect.AbstractMultiset with modifiers "public final"
	at nl.jqno.equalsverifier.integration.extended_contract.ExternalApiClassesTest.succeed_whenClassUsesGoogleGuavaMultiset(ExternalApiClassesTest.java:40)
```

I haven't investigated further yet. May do so later this weekend, but perhaps @jqno instantly sees what's wrong ;)

--- a/src/main/java/nl/jqno/equalsverifier/internal/prefabvalues/factories/AbstractReflectiveGenericFactory.java
+++ b/src/main/java/nl/jqno/equalsverifier/internal/prefabvalues/factories/AbstractReflectiveGenericFactory.java
@@ -15,6 +15,7 @@
  */
 package nl.jqno.equalsverifier.internal.prefabvalues.factories;
 
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import nl.jqno.equalsverifier.internal.exceptions.ReflectionException;
 import nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues;
 import nl.jqno.equalsverifier.internal.prefabvalues.TypeTag;
@@ -71,9 +72,12 @@ public abstract class AbstractReflectiveGenericFactory<T> implements PrefabValue
         return genericTypes.get(n);
     }
 
+    @SuppressFBWarnings(value = "DP_DO_INSIDE_DO_PRIVILEGED", justification = "EV is run only from within unit tests")
     protected void invoke(Class<?> type, Object receiver, String methodName, Class<?>[] classes, Object[] values) {
         try {
             Method method = type.getMethod(methodName, classes);
+            // Not necessary in the common case, but required for https://bugs.java.com/view_bug.do?bug_id=6924232.
+            method.setAccessible(true);
             method.invoke(receiver, values);
         }
         catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {

